Parameters are not properly cleared when selecting a second template from the catalog

Only happens in the template dialog in the new experience.

```
angular.js:14199 Error: [ngRepeat:dupes] Duplicates in a repeater are not allowed. Use 'track by' expression to specify unique keys. Repeater: paramName in $ctrl.parameters.generated, Duplicate key: string:SOURCE_REPOSITORY_REF, Duplicate value: SOURCE_REPOSITORY_REF
http://errors.angularjs.org/1.5.11/ngRepeat/dupes?p0=paramName%20in%20%24ct…eters.generated&p1=string%3ASOURCE_REPOSITORY_REF&p2=SOURCE_REPOSITORY_REF
    at angular.js:68
    at ngRepeatAction (angular.js:30586)
    at $watchCollectionAction (angular.js:17697)
    at Scope.$digest (angular.js:17836)
    at Scope.$apply (angular.js:18102)
    at Array.flushOnChangesQueue (angular.js:8280)
    at Scope.$digest (angular.js:17887)
    at Scope.$apply (angular.js:18102)
    at HTMLAnchorElement.<anonymous> (angular.js:26597)
    at HTMLAnchorElement.dispatch (jquery.js:4435)
```

@jwforres fyi

--- a/app/scripts/directives/processTemplateDialog.js
+++ b/app/scripts/directives/processTemplateDialog.js
@@ -22,7 +22,9 @@ function ProcessTemplateDialog($scope, DataService) {
     id: 'configuration',
     label: 'Configuration',
     view: 'views/directives/process-template-dialog/process-template-config.html',
-    valid: true,
+    // Start initially as invalid so the button doesn't flicker when the dialog
+    // is displayed and the template has required fields.
+    valid: false,
     allowed: true,
     onShow: showConfig
   };
@@ -52,13 +54,29 @@ function ProcessTemplateDialog($scope, DataService) {
     }
   };
 
+  $scope.$on('templateInstantiated', function(event, message) {
+    ctrl.selectedProject = message.project;
+    ctrl.currentStep = ctrl.resultsStep.label;
+  });
+
   ctrl.$onDestroy = function() {
     clearValidityWatcher();
   };
 
-  $scope.$on('templateInstantiated', function(event, message) {
-    ctrl.selectedProject = message.project;
-  });
+  ctrl.next = function(step) {
+    if (step.stepId === ctrl.configStep.id) {
+      instantiateTemplate();
+      // Don't advance wizard automatically. Wait for template validation to complete.
+      return false;
+    }
+
+    if (step.stepId === ctrl.resultsStep.id) {
+      ctrl.close();
+      return false;
+    }
+
+    return true;
+  };
 
   ctrl.close = function() {
     var cb = ctrl.onDialogClosed();
@@ -100,7 +118,6 @@ function ProcessTemplateDialog($scope, DataService) {
     ctrl.resultsStep.selected = true;
     ctrl.nextTitle = "Close";
     clearValidityWatcher();
-    instantiateTemplate();
     ctrl.wizardDone = true;
   }
 

--- a/app/views/_templateopt.html
+++ b/app/views/_templateopt.html
@@ -1,8 +1,7 @@
 <div
     class="template-options"
-    ng-show="parameters.length"
     ng-form="paramForm">
-  <div ng-if="!isDialog" class="flow">
+  <div ng-if="!isDialog && parameters.length" class="flow">
     <div class="flow-block">
       <h2>Parameters</h2>
     </div>

--- a/app/views/directives/process-template-dialog.html
+++ b/app/views/directives/process-template-dialog.html
@@ -6,9 +6,11 @@
        step-class="order-service-wizard-step"
        wizard-ready="$ctrl.wizardReady"
        next-title="$ctrl.nextTitle"
+       next-callback="$ctrl.next"
        on-finish="$ctrl.close()"
        on-cancel="$ctrl.close()"
        wizard-done="$ctrl.wizardDone"
+       current-step="$ctrl.currentStep"
        class="pf-wizard-no-back">
     <div pf-wizard-step ng-repeat="step in $ctrl.steps track by step.id"
          step-title="{{step.label}}"
@@ -41,7 +43,7 @@
           </div>
         </div>
         <div class="order-service-config">
-          <div ng-include="step.view" class="wizard-pf-main-form-contents"></div>
+          <div ng-if="step.selected" ng-include="step.view" class="wizard-pf-main-form-contents"></div>
         </div>
       </div>
     </div>

--- a/app/views/directives/process-template.html
+++ b/app/views/directives/process-template.html
@@ -4,7 +4,7 @@
       <select-project ng-if="!$ctrl.project" selected-project="$ctrl.selectedProject" name-taken="$ctrl.projectNameTaken"></select-project>
     </template-options>
     <!-- TODO: Also support advanced options for new projects. -->
-    <div ng-if="$ctrl.isDialog && $ctrl.selectedProject.metadata.uid && $ctrl.template" class="form-group">
+    <div ng-if="$ctrl.isDialog && $ctrl.selectedProject.metadata.uid" class="form-group">
       <!-- TODO: Preserve entered form values. -->
       To set optional parameters or labels, view
       <a ng-href="{{$ctrl.template | createFromTemplateURL : $ctrl.selectedProject.metadata.name}}">advanced options</a>.

--- a/app/views/landing-page.html
+++ b/app/views/landing-page.html
@@ -9,7 +9,7 @@
       <div class="middle-container">
         <div class="middle-content">
           <overlay-panel show-panel="template" show-close="true" handle-close="templateDialogClosed">
-            <process-template-dialog  template="template" on-dialog-closed="templateDialogClosed"></process-template-dialog>
+            <process-template-dialog ng-if="template" template="template" on-dialog-closed="templateDialogClosed"></process-template-dialog>
           </overlay-panel>
           <landing-page base-project-url="project" on-template-selected="templateSelected">
             <landingsearch>

